### ISSUE TITLE
Fail instead of going into a retry loop

### ISSUE THREAD

**Issue by simonrozsival** [toggl/mobileapp#3759]
During sync it sometimes happen that the server gives us 429 errors and blocks requests for some time. In this case we are supposed to wait for 1 minute and then try again.

On iOS we have only 30 seconds to complete the background task. There is no time for waiting.

If we run into 429, we should simply call it a failed sync and wait for the next slot the OS gives us to try again.

Implementing this will require either hacking the state machine or modifying the way we handle retries. This needs more thought and discussion before we start implementing this feature.

**Comment by simonrozsival** [comment]
The PR was merged into the feature branch.
